pkg/che/che.go

```diff
@@ -23,7 +40,123 @@ type Client struct {
 func InitDefaultCheClient(cfg *crtcfg.Config, cl client.Client) {
 	DefaultClient = &Client{
 		config:     cfg,
+		httpClient: newHTTPClient(),
 		k8sClient:  cl,
 		tokenCache: newTokenCache(),
 	}
 }
+
+// UserExists returns true if the username exists, false if it doesn't and an error if there was problem with the request
+func (c *Client) UserExists(username string) (bool, error) {
+	reqData := url.Values{}
+	reqData.Set("name", username)
+	res, err := c.cheRequest(http.MethodGet, cheUserFindPath, reqData)
+	if err != nil {
+		return false, errors.Wrapf(err, "request to find Che user '%s' failed", username)
+	}
+	defer rest.CloseResponse(res)
+	if res.StatusCode == http.StatusOK {
+		return true, nil
+	} else if res.StatusCode == http.StatusNotFound {
+		return false, nil
+	}
+	return false, errors.Errorf("request to find Che user '%s' failed, Response status: '%s'", username, res.Status)
+}
+
+// GetUserIDByUsername returns the user ID that maps to the given username
+func (c *Client) GetUserIDByUsername(username string) (string, error) {
+	reqData := url.Values{}
+	reqData.Set("name", username)
+	res, err := c.cheRequest(http.MethodGet, cheUserFindPath, reqData)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to get Che user ID for user '%s'", username)
+	}
+	defer rest.CloseResponse(res)
+	if res.StatusCode != http.StatusOK {
+		resBody, _ := rest.ReadBody(res.Body)
```

Comment:
Let's at least log the error from ReadBody()?

Reply:
Agreed, did the same for other areas